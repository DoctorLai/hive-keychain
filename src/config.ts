const Config = {
  hiveEngine: {
    mainNet: 'ssc-mainnet-hive',
  },
  claims: {
    FREQUENCY: +(process.env.DEV_CLAIM_FREQUENCY || 10 * 60 * 1000),
    freeAccount: {
      MIN_RC_PCT: +(process.env.DEV_CLAIM_ACCOUNT_RC_PCT || 95),
      MIN_RC: +(process.env.DEV_CLAIM_ACCOUNT_MIN_RC || 1.1 * 10 * 10 ** 12), // 10% more than 10^13 (current creation cost)
    },
  },
};

export default Config;
