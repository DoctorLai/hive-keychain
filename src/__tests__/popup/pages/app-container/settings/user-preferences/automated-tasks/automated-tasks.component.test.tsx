import { screen } from '@testing-library/react';
import automatedTasks from 'src/__tests__/popup/pages/app-container/settings/user-preferences/automated-tasks/mocks/automated-tasks';
import config from 'src/__tests__/utils-for-testing/setups/config';
config.byDefault();
describe('automated-tasks.component tests:\n', () => {
  let _asFragment: () => DocumentFragment;
  const { methods } = automatedTasks;
  methods.afterEach;
  describe('Stored data:\n', () => {
    describe('Max mana greater than freeAccount credits:\n', () => {
      it.todo('Must load component and match snapshot');
    });
    describe('Max mana lower:\n', () => {
      beforeEach(async () => {
        _asFragment = await automatedTasks.beforeEach({
          passData: true,
          maxManaGreater: true,
        });
      });
      it('Must load component and match snapshot', () => {
        screen.debug();
      });
    });
  });
  describe('No data', () => {
    it.todo('Must load component and match snapshot');
  });
});
