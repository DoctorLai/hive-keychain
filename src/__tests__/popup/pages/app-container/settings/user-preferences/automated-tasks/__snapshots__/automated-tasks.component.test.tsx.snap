// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`automated-tasks.component tests:
 No data Must load component and match snapshot: automated-tasks.component No Data 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Automated tasks
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="automated-tasks-page"
        >
          <div
            class="intro"
          >
            Setup automated tasks that will run every hour if fitting your conditions.
          </div>
          <div
            class="select"
          >
            <div>
              <div
                aria-expanded="false"
                aria-label="Dropdown select"
                class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
                color="#0074D9"
                direction="ltr"
                tabindex="0"
              >
                <div
                  class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
                >
                  <div
                    class="selected-account-panel"
                  >
                    <img
                      src="https://images.hive.blog/u/keychain.tests/avatar"
                    />
                    <div
                      class="selected-account-name"
                    >
                      keychain.tests
                    </div>
                  </div>
                </div>
                <div
                  class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
                  color="#0074D9"
                  rotate="1"
                  tabindex="-1"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 40 40"
                  >
                    <path
                      d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-rewards-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-rewards-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-rewards"
                class="label"
              >
                Claim Rewards
              </div>
            </div>
            <div
              class="hint"
            >
              Regularly claiming your rewards can increase your gains.
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-savings-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-savings-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-savings"
                class="label"
              >
                Claim savings interest
              </div>
            </div>
            <div
              class="hint"
            >
              To claim your savings interest, Keychain will deposit or withdraw 0.001 HBD to/from your savings every 30 days
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`automated-tasks.component tests:
 Stored data:
 Max mana greater than freeAccount credits:
 Must load component and match snapshot: automated-tasks.component With Data Mana Greater 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Automated tasks
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="automated-tasks-page"
        >
          <div
            class="intro"
          >
            Setup automated tasks that will run every hour if fitting your conditions.
          </div>
          <div
            class="select"
          >
            <div>
              <div
                aria-expanded="false"
                aria-label="Dropdown select"
                class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
                color="#0074D9"
                direction="ltr"
                tabindex="0"
              >
                <div
                  class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
                >
                  <div
                    class="selected-account-panel"
                  >
                    <img
                      src="https://images.hive.blog/u/keychain.tests/avatar"
                    />
                    <div
                      class="selected-account-name"
                    >
                      keychain.tests
                    </div>
                  </div>
                </div>
                <div
                  class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
                  color="#0074D9"
                  rotate="1"
                  tabindex="-1"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 40 40"
                  >
                    <path
                      d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-rewards-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-rewards-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-rewards"
                class="label"
              >
                Claim Rewards
              </div>
            </div>
            <div
              class="hint"
            >
              Regularly claiming your rewards can increase your gains.
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-accounts-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-accounts-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-accounts"
                class="label"
              >
                Claim Accounts
              </div>
            </div>
            <div
              class="hint"
            >
              Claim a free account ticket when your RC percentage are over 75%. Depending on your Hive Power, this might use up a big chunk of your RC.
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-savings-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-savings-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-savings"
                class="label"
              >
                Claim savings interest
              </div>
            </div>
            <div
              class="hint"
            >
              To claim your savings interest, Keychain will deposit or withdraw 0.001 HBD to/from your savings every 30 days
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;

exports[`automated-tasks.component tests:
 Stored data:
 Max mana lower:
 Must load component and match snapshot: automated-tasks.component With Data Mana Lower 1`] = `
<DocumentFragment>
  <div
    class="App "
  >
    <div
      class="app-router"
      style="height: 100%; display: grid; grid-template-rows: 70px 1fr;"
    >
      <div
        class="title-section"
      >
        <span
          aria-label="arrow-back-icon"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          arrow_back
        </span>
        <div
          class="title"
        >
          Automated tasks
        </div>
        <span
          aria-label="icon-close-page"
          class="icon-component material-icons-outlined icon-button clickable"
        >
          clear
        </span>
      </div>
      <div
        class="page-content"
        style="overflow: auto; display: flex; flex-direction: column;"
      >
        <div
          class="automated-tasks-page"
        >
          <div
            class="intro"
          >
            Setup automated tasks that will run every hour if fitting your conditions.
          </div>
          <div
            class="select"
          >
            <div>
              <div
                aria-expanded="false"
                aria-label="Dropdown select"
                class="react-dropdown-select select-account-select css-16vukcu e1gzf2xs0"
                color="#0074D9"
                direction="ltr"
                tabindex="0"
              >
                <div
                  class="react-dropdown-select-content react-dropdown-select-type-single css-1m5113o e1gn6jc30"
                >
                  <div
                    class="selected-account-panel"
                  >
                    <img
                      src="https://images.hive.blog/u/keychain.tests/avatar"
                    />
                    <div
                      class="selected-account-name"
                    >
                      keychain.tests
                    </div>
                  </div>
                </div>
                <div
                  class="react-dropdown-select-dropdown-handle css-ago8sv e1vudypg0"
                  color="#0074D9"
                  rotate="1"
                  tabindex="-1"
                >
                  <svg
                    fill="currentColor"
                    viewBox="0 0 40 40"
                  >
                    <path
                      d="M31 26.4q0 .3-.2.5l-1.1 1.2q-.3.2-.6.2t-.5-.2l-8.7-8.8-8.8 8.8q-.2.2-.5.2t-.5-.2l-1.2-1.2q-.2-.2-.2-.5t.2-.5l10.4-10.4q.3-.2.6-.2t.5.2l10.4 10.4q.2.2.2.5z"
                    />
                  </svg>
                </div>
              </div>
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-rewards-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-rewards-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-rewards"
                class="label"
              >
                Claim Rewards
              </div>
            </div>
            <div
              class="hint"
            >
              Regularly claiming your rewards can increase your gains.
            </div>
          </div>
          <div
            class="checkbox-container align-top"
          >
            <div
              class="checkbox-and-label"
            >
              <div
                class="pretty p-default"
              >
                <input
                  aria-checked="true"
                  id="checkbox-autoclaim-savings-inner-input"
                  type="checkbox"
                  value=""
                />
                <div
                  class="state"
                >
                  <label
                    for="checkbox-autoclaim-savings-inner-input"
                  />
                </div>
              </div>
              <div
                aria-label="checkbox-autoclaim-savings"
                class="label"
              >
                Claim savings interest
              </div>
            </div>
            <div
              class="hint"
            >
              To claim your savings interest, Keychain will deposit or withdraw 0.001 HBD to/from your savings every 30 days
            </div>
          </div>
        </div>
      </div>
    </div>
    <div
      class="Toastify"
    />
    <div
      aria-label="proposal-voting-section"
      class="proposal-voting-section hide closed"
    >
      <span
        class="icon-component material-icons close clickable"
      >
        clear
      </span>
      <div
        class="text"
      >
        Support Keychain development proposal
      </div>
    </div>
  </div>
</DocumentFragment>
`;
